extensions/notebook/src/jupyter/jupyterServerInstallation.ts

```diff
@@ -320,6 +320,11 @@ export class JupyterServerInstallation implements IJupyterServerInstallation {
 			}
 		}
 
+		// Delete existing Python variables in ADS to prevent conflict with other installs
+		delete process.env['PYTHONPATH'];
```

Comment:
If we're going to be doing this to the ADS process shouldn't this be something we do at startup? Because otherwise you may be having stuff running along using the env vars happily but then all of a sudden have them removed and break other things once you actually open a notebook.

Reply:
While this may be the case, there are other scenarios that leverage Python that could be broken by making a change globally at startup (ML extension as an example), so I'm hesitant to make a global change like that, since that could impact a number of scenarios outside of notebooks . I'd prefer to keep our existing behavior at the moment until we need to change this to unblock those folks who are hitting this particular bug.

Reply:
But isn't this still going to be a breaking issue for those if that's the case? It'll just happen later instead of at startup.

It seems like we should be starting up the Jupyter instance with those env vars removed instead of modifying the ADS process itself.

Reply:
Only if users open a notebook before leveraging those other scenarios. I agree that it's not perfect for sure.

It's really 2 things:
- Jupyter should be started without these env vars
- There are other Python calls that exist without Jupyter being in the picture (pip/conda install, python user folder, etc). 

The 2nd one is a bit trickier so trying to figure out how we can prevent impacts in other scenarios there. Remove env vars, run whatever command, then set those vars back to their previous values?

Reply:
Would using a virtual environment fix those issues?

Reply:
And for the commands - we're almost always spawning a child process for those so can modify the environment used for that directly.

Reply:
@corivera potentially yes. If we wanted to go down that route in the future, there are a few teams that we would want to talk to. Have heard some stories about trickiness there too 😄

Reply:
https://github.com/microsoft/azuredatastudio/issues/13144

I've added a priority tag there. Merging in this fix for existing behavior is high pri, so we'll go with this for now instead of blocking for a more ideal fix. I'll need to setup a repro environment myself, but after looking through the code, this looks doable; the trick will be ensuring that we catch all of the places where we interact with `child-process`